editor/animation_track_editor.cpp

```diff
@@ -6714,6 +6762,142 @@ void AnimationTrackEditor::_edit_menu_pressed(int p_option) {
 			AnimationPlayerEditor::get_singleton()->get_player()->apply_reset(true);
 		} break;
 
+		case EDIT_CONVERT_TO_BEZIER: {
+			track_convert_select->clear();
+			TreeItem *troot = track_convert_select->create_item();
+
+			for (int i = 0; i < animation->get_track_count(); i++) {
+				if (animation->track_get_key_count(i) == 0) {
+					continue; // Don't list animations with no keys.
+				}
+				NodePath path = animation->track_get_path(i);
+				Node *node = nullptr;
+
+				if (root) {
+					node = root->get_node_or_null(path);
+				}
+
+				String text;
+				Ref<Texture2D> icon = get_editor_theme_icon(SNAME("Node"));
+				if (node) {
+					if (has_theme_icon(node->get_class(), EditorStringName(EditorIcons))) {
+						icon = get_editor_theme_icon(node->get_class());
+					}
+
+					text = node->get_name();
+					Vector<StringName> sn = path.get_subnames();
+					for (int j = 0; j < sn.size(); j++) {
+						text += ".";
+						text += sn[j];
+					}
+
+					path = NodePath(node->get_path().get_names(), path.get_subnames(), true);
+				} else {
+					text = path;
+					int sep = text.find(":");
+					if (sep != -1) {
+						text = text.substr(sep + 1, text.length());
+					}
+				}
+
+				String track_type;
+				switch (animation->track_get_type(i)) {
+					case Animation::TYPE_VALUE:
+						break; // Show only the node path.
+					case Animation::TYPE_POSITION_3D:
+						track_type = TTR("Position");
+						break;
+					case Animation::TYPE_ROTATION_3D:
+						track_type = TTR("Rotation");
+						break;
+					case Animation::TYPE_SCALE_3D:
+						track_type = TTR("Scale");
+						break;
+					case Animation::TYPE_BLEND_SHAPE:
+						track_type = TTR("BlendShape");
+						break;
```

Comment:
They should not be converted for performance reasons and because they perform different calculations than ValueTrack and BezierTrack for RootMotion, etc.

Reply:
Ah I don't think I fully understood these types fully. I made this a simple check so it only allows ValueTrack.

Reply:
The majority of imported animations are TYPE_POSITION_3D, TYPE_ROTATION_3D, TYPE_SCALE_3D and TYPE_BLEND_SHAPE. So won't this result in no conversions?

Reply:
But do people want to convert imported animations to Bezier animations? My usecase and the client I work with is animating in the editor and you start out with regular keys and then you realise you want more control, so you convert that animation into a bezier animation, in my workflows I don't tend to edit imported animations inside Godot.

Reply:
For my use case I want to do bezier key frame reduction.